libs/shinkai-ui/src/components/execution-files.tsx

```diff
@@ -0,0 +1,323 @@
+import { DialogClose } from '@radix-ui/react-dialog';
```

Comment:
@Yiakman shinkai-ui lib is focused on UI components, we avoid adding any server logic there. I think it makes more sense to place this component under tools/components and reuse it where needed.
Also, for previewing files, we already have a component called `FilePreview`. Maybe we could reuse that one here as well?

Reply:
I tried using file preview but it it was too many changes related to the file path and references so i wrote a common component for the execution outside a chat context. I'll see if I can use FilePreview in both chat and execution contexts. For now i moved execution-files to components/tools/components

Reply:
So the main issue was that FilePreview expects file data to be ready right away (like when you attach something to chat), but tool results only give you a file path at first. We need to fetch those tool files when needed using that useGetShinkaiFileProtocol hook. Changing the shared component to handle this special hook-based fetching was tricky, so I just reused the fetching logic we already had in the execution panel

Reply:
@Yiakman just pushed some refactor so we can use same components and logic for file preview

Reply:
awesome